pkg/reconciler/autoscaling/kpa/resources/decider.go

```diff
@@ -43,10 +43,25 @@ func MakeDecider(ctx context.Context, pa *v1alpha1.PodAutoscaler, config *autosc
 			target = annotationTarget
 		}
 	}
+	// Look for a panic threshold percentage annotation.
+	panicThresholdPercentage, ok := pa.PanicThresholdPercentage()
+	if !ok {
+		// Fall back on cluster config.
+		panicThresholdPercentage = config.PanicThresholdPercentage
+	}
+	if panicThresholdPercentage == 0.0 {
+		// Fall back on a hard coded value.
+		panicThresholdPercentage = 200
+	}
```

Comment:
Should these defaults really be applied here? If yes: Have we made sure that a change of the configmap will trigger a reconcile of the decider/metric? (The same applies for MakeMetric).

Reply:
You're right, there is a problem here.  We already have a problem because we read the config here: https://github.com/knative/serving/blob/a3f7c3bba988b812dd4d243b87272bd8c10f5f83/pkg/reconciler/autoscaling/kpa/resources/decider.go#L33 but we don't reconcile deciders when the config maps changes.  We could make the reconciler sensitive to config map changes, but we would have to reconcile *all* deciders.

Let's leave this problem in place.  I will open an issue to tackle the problem in general.  There are probably other places that could benefit from thinking this through carefully.